File: Python3/프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    length = len(stages)
    d = dict()
    for stage in stages:
        if stage in d:
            d[stage] += 1
        else:
            d[stage] = 1
    lst = []
    answer = []
    for i in range(1, N + 1):
        if i in d:
            lst.append((d[i]/length, i))
            length -= d[i]
        else:
            lst.append((0, i))
    
    lst.sort(key=lambda x: (-x[0] ,x[1]))
    for mem in lst:
        answer.append(mem[1])
    return answer

File: JavaScript/프로그래머스/1/42889. 실패율/실패율.js
function solution(N, stages) {
    let answer = new Array(N + 1).fill(0);
    let num = stages.length;
    for (let i = 1; i <= N; i++){
        let count = 0;
        for (let j = 0; j < stages.length; j++){
            if (stages[j] == i) count += 1;
        }
        if (count === 0){
            answer[i] = 0;
        } else {
            answer[i] = count / num;
        }
        num -= count;
    }
    answer.shift();
    const result = Object.entries(answer).sort((a, b) => b[1] - a[1]);
    
    answer = result.map((a) => parseInt(a[0]) + 1);
    return answer;
}